File: backend/app/schemas/transaction.py
from pydantic import BaseModel, Field
from datetime import datetime, date
from decimal import Decimal
from typing import Optional
from uuid import UUID
from .account import Account
from .category import Category

class TransactionBase(BaseModel):
    account_id: UUID
    category_id: UUID
    amount: Decimal = Field(decimal_places=2)
    description: str
    transaction_date: date
    type: str  # "INCOME" or "EXPENSE"

class TransactionCreate(TransactionBase):
    pass

class TransactionUpdate(BaseModel):
    account_id: Optional[UUID] = None
    category_id: Optional[UUID] = None
    amount: Optional[Decimal] = None
    description: Optional[str] = None
    transaction_date: Optional[date] = None
    type: Optional[str] = None

class Transaction(TransactionBase):
    id: UUID
    ai_category_id: Optional[UUID] = None
    ai_confidence: Optional[float] = None
    is_ai_categorized: bool = False
    user_corrected: bool = False
    created_at: datetime
    updated_at: datetime
    account: Optional[Account] = None
    category: Optional[Category] = None
    
    class Config:
        from_attributes = True

File: frontend/src/types/transaction.ts
export interface Category {
  id: string
  name: string
  type: 'INCOME' | 'EXPENSE'
  color?: string
  created_at: string
  updated_at: string
}

export interface Transaction {
  id: string
  account_id: string
  category_id: string
  amount: string
  description: string
  date: string
  type: 'INCOME' | 'EXPENSE'
  ai_category_id?: string
  ai_confidence?: number
  is_ai_categorized: boolean
  user_corrected: boolean
  created_at: string
  updated_at: string
  account?: {
    id: string
    name: string
    account_type?: {
      id: string
      name: string
      category: 'ASSET' | 'LIABILITY'
      sub_category: string
    }
  }
  category?: Category
}

export interface TransactionCreate {
  account_id: string
  category_id: string
  amount: string
  description: string
  date: string
  type: 'INCOME' | 'EXPENSE'
}

export interface TransactionUpdate {
  account_id?: string
  category_id?: string
  amount?: string
  description?: string
  date?: string
  type?: 'INCOME' | 'EXPENSE'
}